### PR TITLE
Additional endpoints, UUID for keys

### PR DESCRIPTION
**Pull request by edreeseg**
Added an endpoint to handle GET operation to both /providers and /immunizations.

All unsigned int IDs have been switched to UUIDs.  Logic has been written into the migration to make use of unsigned ints when the DB is launched with SQLite3, and logic has been written into the seeds to ensure foreign keys are accurate across tables.

### CODE REVIEW

File: api/middleware/authenticate.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 const { jwtSecret } = require('../../config/secrets');
+const { getUser } = require('../../data/helpers');
 
-function authenticate(req, res, next) {
+async function authenticate(req, res, next) {
   try {
     const header = req.get('Authorization');
     if (header) {
       const token = header.split(' ')[1];
+      if (!token) throw new Error('Token must be set on Authorization header.');
```
Comment (Mister-Corn) [inline api/middleware/authenticate.js]: Excellent. We're gonna forget to put that in I promise you.

We also have a similar error in line 22. Is this an unintentional double-up? Or are there two cases you have to handle separately to throw the error?

File: api/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    authRoutes: require('./authRoutes'),
+    userRoutes: require('./userRoutes'),
+    patientRoutes: require('./patientRoutes'),
+    providerRoutes: require('./providerRoutes'),
+    immunizationRoutes: require('./immunizationRoutes'),
+};
```
Comment (Mister-Corn) [inline api/routes/index.js]: Nice way to clean up import statements :heart_eyes:

File: data/migrations/20190423082317_create_users_table.js [hunk 3]
```diff
@@ -1,6 +1,13 @@
 exports.up = function(knex, Promise) {
   return knex.schema.createTable('users', tbl => {
-    tbl.increments();
+    if (knex.client.config.client === 'pg') {
+      tbl
+        .uuid('id')
+        .primary()
+        .defaultTo(knex.raw('uuid_generate_v4()'));
```
Comment (Mister-Corn) [inline data/migrations/20190423082317_create_users_table.js]: I was envisioning the UUID to be generated in the JS files, then inserted into the database. If I understand this correctly, knex can generate UUID's on its own?

File: data/seeds/production/01_users.js [hunk 4]
```diff
@@ -2,89 +2,72 @@ exports.seed = function(knex, Promise) {
...
+          providerId: providers[i].id,
+          createdAt: knex.fn.now(),
+        });
+      }
+      const nonProviders = [
         {
-          id: 1,
-          username: 'edreeseg',
+          username: 'ed_reeseg',
           password: 'test123',
```
Comment (Mister-Corn) [inline data/seeds/production/01_users.js]: Is it possible to log in as seed users despite the passwords here not being hashed?

File: data/seeds/production/02_patients.js [hunk 5]
```diff
@@ -1,130 +1,22 @@
...
+        if (users[i].providerId) continue;
+        const name = users[i].username.split('_');
+        newPatients.push({
+          firstName: name[0],
+          lastName: name[1],
+          birthDate: '1970-01-01',
+          userId: users[i].id,
+          createdAt: knex.fn.now(),
+        });
+      }
```
Comment (Mister-Corn) [inline data/seeds/production/02_patients.js]: :heart_eyes:

**Comment by Mister-Corn** [comment]
@edreeseg I tried to add Travis CI so it can automatically run tests whenever you push, but I'm running into issues. I'll merge this as is, and when I have time, I'll try to see what's up.

File: data/seeds/production/01_users.js (hunk 4 above)
Comment (edreeseg) [reply data/seeds/production/01_users.js]: At the moment it's not possible, but I'm probably going to run the seeds' passwords through a function that'll run bcrypt on them to hash, to make logging in possible.

File: data/migrations/20190423082317_create_users_table.js (hunk 3 above)
Comment (edreeseg) [reply data/migrations/20190423082317_create_users_table.js]: Postgres has a built-in UUID type, and you can use the uuid-ossp module to generate a uuid in this way.

File: api/middleware/authenticate.js (hunk 1 above)
Comment (edreeseg) [reply api/middleware/authenticate.js]: This is something I'll definitely be refactoring to make more DRY.  Essentially the one on line 22 is just a catch-all if, in some alternate universe, the code somehow executes to that point.

**Pull request merged**